src/main/java/com/hackathonorganizer/hackathonwriteservice/team/service/TeamService.java

```diff
@@ -0,0 +1,93 @@
+package com.hackathonorganizer.hackathonwriteservice.team.service;
+
+import com.hackathonorganizer.hackathonwriteservice.hackathon.model.Hackathon;
+import com.hackathonorganizer.hackathonwriteservice.hackathon.repository.HackathonRepository;
+import com.hackathonorganizer.hackathonwriteservice.team.exception.ResourceNotFoundException;
+import com.hackathonorganizer.hackathonwriteservice.team.model.Tag;
+import com.hackathonorganizer.hackathonwriteservice.team.model.Team;
+import com.hackathonorganizer.hackathonwriteservice.team.model.dto.TagRequest;
+import com.hackathonorganizer.hackathonwriteservice.team.model.dto.TeamRequest;
+import com.hackathonorganizer.hackathonwriteservice.team.model.dto.TeamResponse;
+import com.hackathonorganizer.hackathonwriteservice.team.repository.TeamRepository;
+import com.hackathonorganizer.hackathonwriteservice.team.utils.TeamMapper;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class TeamService {
+
+
+    private final TeamRepository teamRepository;
+    private final TagService tagService;
+    //TODO change to service when it will be ready
+    private final HackathonRepository hackathonRepository;
+
+    public TeamResponse create(TeamRequest teamRequest) {
+        val teamToSave = Team.builder()
+                .ownerId(teamRequest.ownerId())
+                .hackathon(getHackathon(teamRequest.hackathonId()))
+                .teamMembersIds(teamRequest.teamMembersIds())
+                .tags(getTags(teamRequest.tags()))
+                .build();
+        log.info("Trying to save new team {}", teamToSave);
+        return TeamMapper.toResponse(teamRepository.save(teamToSave));
+    }
+
+    public TeamResponse editById(Long id, TeamRequest teamRequest) {
+        return teamRepository.findById(id)
+                .map(teamToEdit -> {
```

Comment:
Moze wydzielic ten kod do mappera?

Reply:
Edytowanie? Tu nie ma nic z mappera

Reply:
To moze do prywatnej metody by zwiekszyc czytelnosc kodu?

Reply:
Ale co mam przenieść? To jest cała logika odpowiedzialna za edytowanie. To mam wynieść to do innej metody i jak ją nazwać? Też edit? A w edit tylko prywatne metody wywolywac? Bo nie rozumiem co mam wydzielić

Reply:
tak to edit do prywatnej metody, zeby poprawic czytelnosc kodu

Reply:
mocno przesadzasz z tym wydzielaniem, rozumiem jakby to była metoda na 50 linijek robiąca kilka operacji, wtedy warto każdą z nich faktycznie wydzielić w sensowne kawałki. Bardziej czytelność zmniejsza jak żeby dobić się do logiki musisz się przebić przez 5 jednolinijkowych metod.